TAO/TAO_IDL/ast/ast_map.cpp

```diff
@@ -0,0 +1,349 @@
+      ACE_ERROR_RETURN ((LM_ERROR,
+                         ACE_TEXT ("AST_Map::in_recursion - ")
+                         ACE_TEXT ("bad base type\n")),
+                        false);
+    }
+
+  AST_Decl::NodeType nt = type->node_type ();
+
+  if (nt == AST_Decl::NT_typedef)
+    {
+      AST_Typedef *td = dynamic_cast<AST_Typedef*> (type);
+      type = td->primitive_base_type ();
+      nt = type->node_type ();
+    }
+
+  if (nt != AST_Decl::NT_struct
+      && nt != AST_Decl::NT_union
+      && nt != AST_Decl::NT_valuetype
+      && nt != AST_Decl::NT_map)
+    {
+      return false;
+    }
+
+  bool recursion_found = false;
+  AST_Type** recursable_type = nullptr;
+  list.get (recursable_type, 0);
+  if (!ACE_OS::strcmp (type->full_name (),
+                           (*recursable_type)->full_name ()))
+    {
+      // They match.
+      recursion_found = true;
+      idl_global->recursive_type_seen_ = true;
+    }
+  else
+    {
+      // Check the element type.
+      recursion_found = type->in_recursion (list);
+    }
+
+  return recursion_found;
+}
+
+// Redefinition of inherited virtual operations.
+
+// Dump this AST_Map node to the ostream o.
+void
+AST_Map::dump (ACE_OSTREAM_TYPE &o)
+{
+  this->dump_i (o, "map <");
+  AST_Annotation_Appls::iterator i,
+    finished = key_type_annotations ().end ();
+  for (i = key_type_annotations ().begin (); i != finished; ++i)
+    {
+      AST_Annotation_Appl *a = i->get ();
+      a->dump (o);
+      dump_i (o, " ");
+    }
+  this->key_pd_type->dump (o);
+  this->dump_i (o, ", ");
+
+  finished = value_type_annotations ().end ();
+  for (i = value_type_annotations ().begin (); i != finished; ++i)
+    {
+      AST_Annotation_Appl *a = i->get ();
+      a->dump (o);
+      dump_i (o, " ");
+    }
+  this->value_pd_type->dump (o);
+  this->dump_i (o, ", ");
+  this->pd_max_size->dump (o);
+  this->dump_i (o, ">");
+}
+
+int
+AST_Map::ast_accept (ast_visitor *visitor)
+{
+  return visitor->visit_map (this);
+}
+
+// Data accessors.
+
+AST_Expression *
+AST_Map::max_size ()
+{
+  return this->pd_max_size;
+}
+
+AST_Type *
+AST_Map::key_type () const
+{
+  return this->key_pd_type;
+}
+
+AST_Type *
+AST_Map::value_type () const
+{
+  return this->value_pd_type;
+}
+
+AST_Type *
+AST_Map::primitive_key_type () const
+{
+  AST_Type *type_node = key_type ();
+  if (type_node && type_node->node_type () == AST_Decl::NT_typedef)
+    {
+      AST_Typedef *const typedef_node = dynamic_cast<AST_Typedef *> (type_node);
+      if (!typedef_node) return nullptr;
+      type_node = typedef_node->primitive_base_type ();
+    }
+  return type_node;
+}
+
+AST_Type *
+AST_Map::primitive_value_type () const
+{
+  AST_Type *type_node = value_type();
+  if (type_node && type_node->node_type() == AST_Decl::NT_typedef)
+    {
+      AST_Typedef * const typedef_node = dynamic_cast<AST_Typedef*>(type_node);
+      if (!typedef_node) return nullptr;
+      type_node = typedef_node->primitive_base_type();
+    }   
```

Comment:
```suggestion
    }
```

One of the trailing whitespace that causing the fuzz test to fail.